chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/FuzzyChoicesDAOImpl.java

```diff
@@ -0,0 +1,120 @@
+public class FuzzyChoicesDAOImpl extends BaseDAOImpl implements FuzzyChoicesDAO {
+
+    @Transactional
+    public FuzzyChoicesDTO create(FuzzyChoicesDTO dto)
+        throws EntityRetrievalException, EntityCreationException, JsonParseException, JsonMappingException, IOException {
+        FuzzyChoicesEntity entity = new FuzzyChoicesEntity();
+        String toJSON = new ObjectMapper().writeValueAsString(dto.getChoices());
+        entity.setChoices(toJSON);
+        entity.setFuzzyType(dto.getFuzzyType());
+        entity.setCreationDate(new Date());
+        entity.setDeleted(false);
+        entity.setLastModifiedDate(new Date());
+        entity.setLastModifiedUser(Util.getCurrentUser().getId());
+        create(entity);
+        return new FuzzyChoicesDTO(entity);
+    }
+
+    public FuzzyChoicesDTO getByType(FuzzyType fuzzy) throws EntityRetrievalException, JsonParseException, JsonMappingException, IOException {
+        FuzzyChoicesEntity entity = getEntityByType(fuzzy);
+
+        FuzzyChoicesDTO dto = null;
+        if (entity != null) {
+            dto = new FuzzyChoicesDTO(entity);
+        }
+        return dto;
+    }
+
+    @Override
+    @Transactional
+    public List<FuzzyChoicesDTO> findAllTypes() throws EntityRetrievalException, JsonParseException, JsonMappingException, IOException {
+        List<FuzzyChoicesEntity> entities = entityManager
+            .createQuery("SELECT fuzzy FROM FuzzyChoicesEntity fuzzy WHERE (fuzzy.deleted <> true) ",
+                         FuzzyChoicesEntity.class)
+            .getResultList();
+
+        List<FuzzyChoicesDTO> dtos = new ArrayList<FuzzyChoicesDTO>();
+        for (FuzzyChoicesEntity entity : entities) {
+            FuzzyChoicesDTO dto = new FuzzyChoicesDTO(entity);
+            dtos.add(dto);
+        }
+        return dtos;
+    }
+
+    @Transactional
+    public FuzzyChoicesDTO update(FuzzyChoicesDTO dto)
+        throws EntityRetrievalException, EntityCreationException, JsonParseException, JsonMappingException, IOException {
+        FuzzyChoicesEntity entity = getEntityByType(dto.getFuzzyType());
+        if (entity == null) {
+            throw new EntityRetrievalException("Cannot update entity with type " + dto.getFuzzyType() + ". Entity does not exist.");
+        }
+
+        String toJSON = new ObjectMapper().writeValueAsString(dto.getChoices());
+        entity.setChoices(toJSON);
+        entity.setLastModifiedUser(Util.getCurrentUser().getId());
+
+        update(entity);
+        return new FuzzyChoicesDTO(entity);
+    }
+
+    private FuzzyChoicesEntity getEntityByType(FuzzyType type)
+        throws EntityRetrievalException {
+
+        FuzzyChoicesEntity entity = null;
+
+        String queryStr = "SELECT fuzzy from FuzzyChoicesEntity fuzzy where "
+            + "(fuzzy_choices_id = :entityId)";
+
+        Query query = entityManager.createQuery(queryStr, FuzzyChoicesEntity.class);
+        query.setParameter("entityId", (type.ordinal()+1));
```

Comment:
I don't think this is quite the right way to do this... it requires your java enum values to be in the same order as your database enum values, which they are so it will work but.... . Do you have to get the fuzzy type by ID? Maybe you could use the name?